source_pubsub/source.go

```diff
@@ -78,18 +78,10 @@ func (s *Source) Open(ctx context.Context, sdkPos sdk.Position) error {
 		Str("topics", strings.Join(s.config.TopicNames, ",")).
 		Msg("Open Source Connector")
 
-	parsedPositions, err = position.ParseSDKPosition(sdkPos)
+	parsedPositions, err = position.ParseSDKPosition(sdkPos, s.config.TopicName)
```

Comment:
Do you want to pass all the topics here?

Reply:
We only need to pass the old config to process the old config scenario, if multiple topics are being used we parse as normal

Reply:
@anna-cross Oh right, but what if you have new config with old position?

Reply:
You mean if they are updating the config to use new one? I dont believe we allow that at the moment, if a person is using old config and the pipeline is deployed they cant switch over to new one

Reply:
Yes, I'm referring to this in general. If you get the new config with an old position this may not work as expected?

The position which is passed to `Open(..)` is the last position committed to Conduit, if you start the connector and the position is upgraded but not committed, next time it will come up with the same position. While this may not happen with mdpx right now, it can turn into an edge case later is what I'm saying ;-)